### ISSUE TITLE
Downloads folder internationalization

### ISSUE THREAD

**Issue by feross** [feross/webtorrent-desktop#359]
*Moved from: https://github.com/feross/webtorrent/issues/729#issuecomment-207133390* (@nkour)

Apparently the folder name can be different if the OS language is different? How do we detect this?

**Comment by yciabaud** [comment]
I believe that you have to use a windows env var like CSIDL_DEFAULT_DOWNLOADS https://technet.microsoft.com/en-us/library/cc749104(v=ws.10).aspx

**Comment by yciabaud** [comment]
I was wrong, this variable is not accessible in windows env. 

There is a registry key in `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Shell` Folders named `{374DE290-123F-4565-9164-39C4925E467B}` that contains the path to the download folder.

**Comment by ChrisMorrisOrg** [comment]
Not very elegant, but I suppose you could detect if every possible downloads folder exists and pick the first one as the default? :/

Really need that preferences window!

**Comment by nkour** [comment]
doesn't have to do with i18n, in windoz u can always change the destination folder for Downloads, Music etc

**Comment by feross** [comment]
@nkour An option to set your downloads folder will come once there's a Preferences/Setting window.

This issue is about the default downloads folder. Is yours called "Downloads" or something else?

**Comment by nkour** [comment]
@feross it seems you never used windoz :D  How it shows to the user (i18n) is irrelevant since the path is always latin.

I'm talking about redirection in the OS settings: http://windows.microsoft.com/en-us/windows/redirect-folder-new-location#1TC=windows-7

**Comment by feross** [comment]
@nkour Okay, so you have a "redirected" Downloads folder. What happens right now? Where does WebTorrent save files for you?

**Comment by nkour** [comment]
@feross it's not respected as I've stated in https://github.com/feross/webtorrent/issues/729#issue-146715143

**Comment by feross** [comment]
@nkour I need you to be a bit more specific than that. Which folder is the file getting saved to?

**Comment by sindresorhus** [comment]
Just use the Electron API:

```js
require('electron').app.getPath('downloads');
```

https://github.com/electron/electron/blob/master/docs/api/app.md#appgetpathname

**Comment by yciabaud** [comment]
Nicer!

**Comment by grunjol** [comment]
Someone asked the same question yesterday
#349

**Comment by Tercus** [comment]
> If the folder is redirected, and WebTorrent is not putting files there, then where is it putting them?
> 
> Which folder are files getting saved to?

It creates a folder in "C:\User\<name>\Downloads". Normally that is a default folder in Windows for downloads (is setup in the explorer as Favorite), but as has been mentioned, it can be changed.
For example I set mine to be "D:\Downloads" and I deleted the "C:\User\<name>\Downloads" folder. After using the webtorrent-desktop app, that folder was back again containing the files I downloaded
